mock-api/affinity-group.ts

```diff
@@ -0,0 +1,198 @@
+ */
+
+import type {
+  AffinityGroup,
+  AffinityGroupMember,
+  AntiAffinityGroup,
+  AntiAffinityGroupMember,
+} from '@oxide/api'
+
+import { instance, instanceDb2, startingInstance } from './instance'
+import type { Json } from './json-type'
+import { project } from './project'
+
+export const affinityGroup: Json<AffinityGroup> = {
+  id: '3ee49935-b1db-45dd-bfda-67905e6bcab9',
+  name: 'main-web-prod',
+  description: 'Main web deployments',
+  failure_domain: 'sled',
+  policy: 'allow',
+  project_id: project.id,
+  time_created: '2025-03-01T00:00:00.000Z',
+  time_modified: '2025-04-05T12:10:07.917Z',
+}
+
+export const sandboxAffinityGroup: Json<AffinityGroup> = {
+  id: 'c5866165-0fd9-477a-b5ee-0094eaa93dab',
+  name: 'sandbox-env',
+  description: '',
+  failure_domain: 'sled',
+  policy: 'fail',
+  project_id: project.id,
+  time_created: '2025-03-03T00:00:00.000Z',
+  time_modified: '2025-03-09T00:00:00.000Z',
+}
+
+export const affinityGroups: Json<AffinityGroup>[] = [
+  affinityGroup,
+  sandboxAffinityGroup,
+  {
+    id: '3989818a-6d82-4715-b486-189c981f2c0a',
+    name: 'test',
+    description: '',
+    failure_domain: 'sled',
+    policy: 'fail',
+    project_id: project.id,
+    time_created: '2025-04-13T00:00:00.000Z',
+    time_modified: '2025-04-16T00:00:00.000Z',
+  },
+]
+
+export const affinityGroupMemberInstance: Json<AffinityGroupMember> = {
+  type: 'instance',
+  value: { id: instance.id, name: instance.name, run_state: 'stopped' },
+}
+
+export const affinityGroupMemberStartingInstance: Json<AffinityGroupMember> = {
+  type: 'instance',
+  value: {
+    id: startingInstance.id,
+    name: startingInstance.name,
+    run_state: startingInstance.run_state,
+  },
+}
+
+export const nonMatchingAffinityGroupMemberInstance: Json<AffinityGroupMember> = {
+  type: 'instance',
+  value: {
+    id: '4f1f4dc9-da8a-48d6-ace2-c3f6330d4a5e',
+    name: 'a non-matching instance that shouldn’t be in the list',
+    run_state: 'stopped',
+  },
+}
+
+export const affinityGroupMembers: Json<AffinityGroupMember>[] = [
+  affinityGroupMemberInstance,
+  affinityGroupMemberStartingInstance,
+  nonMatchingAffinityGroupMemberInstance,
+]
+
+type DbAffinityGroupMember = {
+  affinity_group_id: string
+  affinity_group_member: Json<AffinityGroupMember>
+}
+
+// Although these are properly constructed, the antiAffinityGroupMemberLists function in handlers.ts
+// will dynamically construct the affinity / anti-affinity group members based on the instances in
+// the instance list, using the ID of the affinity_group_member
+export const affinityGroupMemberLists: DbAffinityGroupMember[] = [
+  {
+    affinity_group_id: affinityGroup.id,
+    affinity_group_member: affinityGroupMemberInstance,
+  },
+  {
+    affinity_group_id: affinityGroup.id,
+    affinity_group_member: affinityGroupMemberStartingInstance,
+  },
+  {
+    affinity_group_id: sandboxAffinityGroup.id,
+    affinity_group_member: affinityGroupMemberInstance,
+  },
+]
```

Comment:
You can cut a bunch of lines in this file by deleting `affinityGroupMemberInstance` and friends and changing the join list to only use the instance ID.

```ts
type DbAffinityGroupMember = {
  affinity_group_id: string
  affinity_group_member: { type: 'instance'; id: string }
}
```